### DISCUSSION TITLE
Suggestion Regarding the (Re)Implementation of “Weak” Objects

### DISCUSSION THREAD

**Discussion by Goatlov3r** [ppy/osu#26753] [Ideas]
Hello and sorry in advance if I mess something up, this is my first time ever using GitHub. 

This is a collaborative post between me and VINXIS, with him having written the short, concise TL;DR, while I wrote most of the giant essay. This post was originally going to tackle some of the issues present under the removal of slider leniency, and propose a solution we believed would satisfy everyone. We are in a way still doing that, but this time with a larger overall vision in mind, arguing in favor of adding “weak” objects (sliders AND circles) into the game, so objects containing relaxed, binary judgement / objects that are lenient. Slider heads in osu!stable with ScoreV1 already work like this, so technically those would simply be re-added, but there would also be the brand new addition of weak circles as well. 

VINXIS is not checking this thread, but I will be taking a look at it periodically and attempting to respond to any questions people might have about our suggestion. However, please do keep in mind that I am currently quite busy, so my replies may not always be immediate. 

---

**TL;DR (by VINXIS)** 

Essentially, a per-object toggle for objects (sliders and circles), BUT with them having a clear visual distinction so that it’s apparent they have one type of hit judgement or the other (sv1 circle/sv1 slider head hit judgement). Something like a glow for “strong sliders” and dotted borders for “weak circles” as spitball visual distinction ideas. 

Possibly include a mod that converts objects into either hit judgement type. Since classic and strict tracking exist anyway, one converting objects to either hit judgement in this context would not hurt since it's similar in the level of base mechanics being altered. 

With this, retroactive support in Default gameplay would exist for previously submitted beatmaps that were made without the environmental context of slider heads having circle-like hit accuracy judgement. Mappers would also be able to utilize more tools by having the toggle option for objects (instead of losing them due to the constraint of more simplified game mechanics). Players will also have added abilities in being able to play the game with either hit judgement type for objects. 

---

**Long Essay (by me):**

INTRODUCTION

Hello. With osu!lazer taking over as the main osu! client, many changes are coming to the way sliders work, with one of them being the addition of slider head accuracy, aka the removal of slider leniency. This essentially turns sliders from a “weak” object into a “strong” object, like circles. I have certain concerns I would like to bring up regarding the implementation of this feature. I will not attempt to argue against slider head accuracy / the existence of strong sliders as a whole, but rather I will try to simply highlight some of the shortcomings of a system with no weak objects, and explain what I believe should be done to create the largest possible positive impact onto osu! and its community. I also want to apologize in advance for making this post so late, possibly merely days before osu!lazer allows for ranked score submission. 

To explain my point, I will analyze the usage of a weak object that everyone is familiar with: the slider (and specifically the slider head) in osu!stable when using ScoreV1. 

WEAK OBJECTS (SLIDER LENIENCY) IN MAPPING

When creating a map and choosing which objects to use in any given section, one of the main differences between circles and sliders that mappers have always been aware of is the fact that the latter lack any sort of accuracy judgement, i.e. they are lenient, or “weak”. They can be tapped considerably earlier or later than the intended timing, and still not lead to the player being punished. For years, mappers have used this to their advantage, utilizing it in ways which often go unnoticed by the players, but which are undoubtedly extremely important for the final result, in terms of the created map and the player experience it provides. 

Slider heads have historically been used to make certain sections easier, to allow for a small break, both in terms of physical strain but also in terms of mental focus. They can give the player a moment to breath, to blink, to relax, and to prepare for the harder sections that might be coming up next. They can be used to introduce the player to a concept, a gimmick, a rhythm, without overwhelming them or punishing them for not immediately understanding it. All of this is even more applicable in polyrhythm / multi-bpm timing, where slider heads play an essential role in creating a balanced and consistently difficult map, without ridiculous spikes or unnecessary player frustration. 

Slider heads are more forgiving, contrasting greatly with circles (a “strong” object) in that regard. This contrast has been utilized by mappers in song representation, having proven itself a valuable tool when attempting to place emphasis on certain sounds (and thus objects or patterns) more than others. It has been used to create a difference in atmosphere, vibe, and tone, between a section filled with comfortable sliders, versus one dominated by menacing circles. 

Slider heads radiate a sense of certainty, while circles fill the player with uneasiness, anxiety and worry. This has been subconsciously ingrained into the minds of players after hundreds or perhaps thousands of hours, and can be used to curate and refine their experience with a map even further, down to guiding and leading their cursor movement by understanding where and how they might wish to aim or tap when their muscle memory is taking over, when they are no longer consciously thinking, when they are in the zone. 

Sliders and circles additionally have a sort of balance that puts them at an equilibrium, in terms of both impact, and difficulty. Sliders usually (but not always) have an additional aspect of tapping and aiming to them (needing to hold the key down and release it, and also aim the slider body), but that is countered by their leniency in terms of timing. On the other hand, circles are simpler and only require the initial aiming and tapping to be completed, but they are not lenient. Both objects fulfill their own separate function, Sliders are not just harder circles. They instead have no leniency, but also require more actions after the initial activation. Their focus is more on the “hold” and “release”, contrasting with the circles that focus on the “click”. They are fundamentally a different object, with different properties, that feels and acts differently, and thus generates a very different effect in every way. 

The sudden addition of slider head accuracy (and thus the removal of all weak objects), without any attempts to integrate it more smoothly into the game, will have an immense negative effect on the hundreds of thousands of already existing maps, the scale of which is hard to overstate. All maps up until this point in time have been made with slider leniency in mind, with weak and strong objects together having been utilized in various ways, both to polish and adjust the experience the maps are attempting to provide to the players, as well as to represent the song they accompany as accurately as possible. This is all now being suddenly changed. An integral part behind the design of these works of art is being torn away. 

And of course, future maps are affected too. Current and future mappers will no longer have these tools available to them, to create the maps they desire. They are instead constrained further, with their imagination and creativity greatly limited. Mapper agency is threatened. The state of mapping as a whole is potentially at stake, with a great deal of complexity and nuance being stripped away. An entire form of art is being reduced and simplified for the sake of consistency between objects, or due to puritanical ideals that have plagued the rhythm game genre since its inception. 

THE SUGGESTION

The suggestion I propose is the following: The creation of a per-object toggle for leniency, which, when turned off, will make the object “strong” (possible to get a 100/50 on due to tapping early/late, like circles in all current versions of osu!), but when turned on will make the object “weak” (no accuracy judgement other than 300 possible due to tapping reasons, like sliders in current osu!stable ScoreV1). This would be available for mappers to use on whichever objects they want (sliders AND circles), and would also have a visual indication, so that those objects could easily be distinguished by players during gameplay. Furthermore, I suggest the creation of certain mods to toggle this on/off for various objects (the Classic mod already does this, turning it on for sliders and off for circles). 

All maps that have been created up until this point in time will simply be maps that use only weak sliders and strong circles, which isn't a big deal. There are already tons of maps in the game that just don't use all currently available mapping features or objects, such as maps with no reverse arrows, maps without spinners, maps without sliders or circles, etc. All maps up until 2024 will simply be maps without any strong sliders or weak circles, while maps from now on could use whatever combination of features or objects they want (including or excluding the new ones). 

Regarding the visual distinction between the objects, the possibilities are endless. Weak circles should probably appear a bit dimmed out, be black and white / be greyed out, or have a dotted outline (my personal favorite idea). Strong sliders could have a glow, a glint, an animated shine or sparkle effect, some sort of blinking animation, something special on their outline (flower petals have been used in skins before as part of the hitcircle overlay and look cool), a special color-changing approach circle, whatever. Just something to highlight them slightly, but without making them distracting. Ideally these elements should also be skinnable, similar to hitcircle.png or sliderstartcircle.png, etc. Strong circles and weak sliders will just remain regular, the same way they have looked up until now. 

In terms of mods, we would simply get something similar to the Classic mod, which adds leniency to all sliders (makes all sliders weak) and removes leniency from all circles (makes all circles strong), but for other individual or combined slider and circle leniency toggles. This could also just be done in a single mod, with adjustable options for both sliders and circles within the same menu. With weak and strong objects being tracked separately for each map, this would also allow for some interesting dynamic scoring multipliers that could be implemented in the future. For example, mods that turn objects weak could have lower multipliers in polyrhythm maps that would otherwise require very complex, skilled tapping, but were trivially easy without accuracy judgements other than 300s (e.g. 0.75x). Meanwhile, on a map that already had mostly weak objects, a mod like that could have a higher multiplier (e.g. 0.98x). 

All in all, I feel like this suggestion being implemented can only be positive for the future of the game. I believe a partial implementation, with at least weak sliders being brought back via a toggle, is absolutely a necessity, but weak circles would also be very easy to add and I feel like there isn't really a reason not to go all the way with something like this. Mappers will be given more tools to play around and experiment with, and they will be able to really fulfill their vision for the maps they wish to create, with considerably fewer limitations than ever before. Players will also be given more ways to enjoy the game, being able to choose whether they want to prioritize fun, experiencing the artistic creations the game has to offer, maximum performance points and ranked score, having their rhythm ability challenged, etc. Even skinners will have something new to pour their creativity into. And everything is entirely opt-in, so while tons of members of the community are satisfied, none have any reason to be dissatisfied.

**Comment by peppy** [comment]
I don't personally like this direction. It goes against my idea of keeping things simple for players.

At the end of the day any changes to hitobjects will require my agreement so I don't see this happening.

**Reply by Goatlov3r** [reply]
Thank you for the quick response. 

I understand your concerns. While I do not share your hesitancy, I believe it to be justified to a degree, at least regarding the implementation of weak circles. When it comes to weak sliders however, I see no reason for them not to be reimplemented. Sure, making the game more complex may not be desired, but that doesn't mean it should be made *less* complex either. As things stand right now, lazer's base gameplay and mapping is less complex than that of stable, as the latter had both strong and weak objects that could be utilized and interacted with in various ways, while the former only has strong ones. 

Weak circles may be uncharted territory, but weak sliders have been a staple of osu! since its inception, and I do not believe anyone has ever found them “too complex” to understand, or thought them to otherwise be taking away from the simplicity of the game in any way. I don't see why mappers should from this point onwards have fewer tools at their disposal than they did before. I understand that you don't wish to give them *more* tools, but don't see the point in taking away the ones they already had the entire time. Simply keeping the number of tools the same would suffice. 

This could be achieved via the addition of the proposed toggle and visual distinction I suggested, but only for sliders, rather than for both sliders and circles. This way slider head accuracy is implemented for those who want to use it during mapping or gameplay for whatever reason, but slider leniency is also kept in place to be utilized as it always has. Both strong and weak objects continue to exist in the game, rather than one of them being inexplicably removed. The game still remains very simple at a fundamental level, but mapper expression and agency is preserved, and so is the player experience that mappers have managed to create over the years. A mod that makes all sliders strong could be added as well, in that sense acting oppositely to the Classic mod which renders all sliders weak, so that the mapper's choice can be overridden by that of the player if desired.

**Reply by Goatlov3r** [reply]
Hello, I'm not sure exactly how GitHub works but I want to "open" this suggestion again, since I didn't receive a reply. I also realize that my wording may not have been the clearest, so I want to clarify exactly what I'm suggesting as the best solution that would completely eliminate any issues both sides have with slider accuracy/leniency. It's a slightly tweaked version of the thing I suggested above (in my reply, not in the original post, I have given up on the lenient circles and the per-object toggles, don't worry), and I believe it to be honestly perfect and also trivial to implement. Allow me to explain: 

There should be a toggle inside the map settings where mappers can select whether they would prefer their map to be played with slider accuracy enabled or without. All maps made up until this point would automatically have this set to no slider accuracy (as they were created with that in mind), but this could be changed at the request of the mapper (without unranking etc, hopefully this is possible, perhaps it would require the toggle to be on the website similar to a map description rather than in the editor). And for new maps there would be the option to toggle between slider accuracy and slider leniency. 

At the same time, the player would also be able to choose whether they want slider accuracy or slider leniency, via mods. This could be done in a separate mod, or perhaps it could be part of the Classic mod, the exact implementation doesn't matter too much. 

When a player loads into a map, and their choice and the mapper's choice are the same (they have both chosen slider leniency, or they have both chosen slider accuracy), then nothing additional happens. But if the player has chosen to play the map in a way not intended by the mapper, then there is a pop up (similar to the one in stable about the map being qualified or loved) letting the player know that it's recommended to play this map with slider leniency or slider accuracy, whichever one the mapper has chosen. If the player wants they can go back to the menu and do that and play the map as it was meant to be played, or they can also just ignore the message and play the map in the way they prefer. 

I genuinely see nothing wrong with this suggestion. The past work of mappers is preserved and new players that try out older maps will get to experience them as they were meant to be experienced (if they care about doing that), but also players and mappers who prefer accuracy judgments on sliders can have their way too. The per map toggle set by the mapper is merely a recommendation and can be ignored if the player wishes, similarly to how the player could alter the intended AR or CS of the map, or enable HR and flip the map around, etc. Similar to those stats, there could be a little icon indicating whether a map is recommended to be played with or without slider leniency, and it could be tinted red or something if you have the opposite selected when you're viewing it in song select, so you don't have to go into the actual map to find out. But yea, ultimately, everyone is given the most amount of agency possible. The game is also not made any more complicated than it currently is (by default in stable you have slider leniency but with a mod you can have slider accuracy, and vice versa for lazer, and this is maintained with this suggestion, both exist in the game and are toggled via a mod), and the implementation should also be relatively easy (a mod or even just a setting for a mod, and then a website or editor toggle). 

Please consider this suggestion, and if you still don't consider it worth adding, please explain why so I can tweak it further. The entire style of some mappers relies on slider leniency being a thing, and it would be sad for a newer player to experience those maps and have the experience made duller due to a toggle that already exists (within Classic mod) but they are unaware that they should probably be using to fully understand what the mapper's intention was. At the same time, players who want to challenge their accuracy skill further (or who find slider leniency confusing and overly complicated) should be allowed to play the game in their own way too. This has been a very discussed topic, with entire videos made about it, prominent community members / mappers / BNs quitting the game over it, polls and discussions threads around it made on a weekly basis on /r/osugame, etc etc, and I think this suggestion would genuinely satisfy everyone at the same time, while also being seemingly easy to implement.

**Reply by Goatlov3r** [reply]
Also on second thought, a new mod is not even required, there should just be the toggle by the mapper and then a little pop up that says "For the fullest experience, the creator of this beatmap recommends using the Classic mod, with the Slider Leniency option enabled", or "For the fullest experience, the creator of this beatmap recommends disabling the Classic mod" depending on the circumstance. And for the first case (the mapper suggesting Classic mod), if the player cares about score, or pp, or wants a challenge, or just doesn't want to bother with that stuff, they can ignore it completely, but if they are more interested in the artistic side of mapping they can play the map according to the mapper's recommendation. This is similar to how some maps say "Recommended 50% Background Dim, Storyboard Enabled, Beatmap Skins and Hitsounds Enabled", and you can actually do that if you want the most satisfying experience, or the closest one to what they intended, or you can also just play the map without paying attention to any of that. The main issue is basically just that new players will be missing out on part of the enjoyment that all past maps provide, because they will be lacking this one feature that a lot of them had utilized in some way, and that they won't even know that they are missing out on it. This suggestion is just for a way of letting them know, and then leaving the choice up to them.

**Comment by eclecticfluff** [comment]
I think it would also be relevant to look at other rhythm games with similar note types. Both Chunithm and Maimai have Ex notes which work in a similar way to the proposed weak notes. As such, if going through with this, it may be a good idea to reference the way these note types are distinguished and taught if that is the direction taken. However, I think catch notes may be better.  
In almost every rhythm game with some emphasis on aim, there are catch or other equivalent notes. Prominent examples include Deemo, Cytus, Phigros, Ongeki, Lanota, Wacca, Arcaea and Sound Voltex. Notable exceptions include Nostalgia, jubeat, and Project Sekai. I don’t think the prevalence of this note type is a coincidence, and there are myriad examples of them being put to good use in ways that would be cumbersome at best with traditional taps and holds.

**Reply by Goatlov3r** [reply]
I am not at all familiar with other rhythm games so this isn't something I can comment on personally, but I would be extremely interested in seeing some discussion about it from other community members who have the required experience. Specifically I would be curious to hear how these notes are thought of in terms of their complexity. Does their existence render the game not simple enough for players? This seems to be the main argument against their inclusion in osu!, that they would be too overwhelming or complicated for players and that things should be kept “simple” instead. I would also love to see some examples of how these might be used in mapping (or the equivalent term for level creation in each game), with some explanations for people like me who may lack the familiarity with the various games but still wish to understand. I'm sure there are some interesting applications that take full advantage of the characteristics of these notes and really allow the mappers' creativity to flourish, potentially in ways that just aren't feasible in osu! due to the constraints of the object types we are currently limited to.

**Reply by eclecticfluff** [reply]
Sorry for the delay in responding, it took some time to find and explain some good examples.  
When learning catch notes, new players often tap them like any other notes, but intuitively transition to holding when they repeat inhumanly fast. Even at the highest levels, tapping catch notes when they arrive is sometimes done to keep time on less sparse sections. Thus, even without any explanation of what a tap note is, inexperienced players will find themselves intuitively playing catch notes in a way not detrimental to their improvement. As for how they actually feel to play, it varies.  
Since I know the most about Arcaea, I’ll take my examples from there. Rise of the World charts similar sections both with short arc (hold in the air) notes and tap notes, which are contrasted back to back at 2:06. The former feels calm and airy, but more technical while the latter is straightforward, intense, and stamina draining. Another interesting example of the contrast of having timing or not on notes is seen in Primitive Lights. There are two distinct aim heavy sections. The fist one, which shows up at around 22 seconds uses arc leniency to keep the focus on aim while staying flowy. The second comes at 1 minute 15 seconds which uses tap notes to give the arcs a timing judgement. This, in combination with the slams, makes the section feel much more choppy which matches with the music. In addition, having head accuracy requires the entire arc to be played more accurately since you can’t abuse the early window with impunity. As a final example, in ω4, since there is nothing but vocals to follow at the beginning, keeping to objects without timing judgement both keeps the energy sufficiently low and circumnavigates the issue of there not being clear transients to time taps to. Not shown here, since it can be done just as well with high tick rates, but rapidly repeating catch notes are a common choice for charting synth growls.

Rise of the World: https://youtube.com/watch?v=UaBYyTeuQ_w&t=2m06s
Primitive Lights: https://youtube.com/watch?v=93tuw4143RQ
ω4: https://youtube.com/watch?v=qpulQds2AiY

**Reply by LocalGuard039** [reply]
But isn't this catch note sounds familiar, like slider ticks or fast reverse sliders? Unlike most of the rhythm games, osu standard have only one note one time, nor have a clear line or make catch note on our way because only one hitable object exist.

**Reply by eclecticfluff** [reply]
The lack of a judgement line is far from unique. Sparkline, Tone sphere, Lyrica, jubeat, and others I can’t remember off the top of my head all have gameplay very similar to osu. The fact that there is only one hit object at a time is an apreciable difference though. ONGEKI, and to a lesser extent Rotaeno all have a singular analog input like osu. They both heavily feature catch notes relying on that analog input. Admittedly these still aren’t the most similar, but I hope it shows that osu isn’t so different from other traditional rhythm games that it can’t take reference from them on gameplay.  
Discreet hit-objects, even if physically played the exact same way as a slider feel different. I don’t know much about osu mapping culture, but from what little I’ve seen, it looks like buzz sliders are typically reserved for fast kicks, and dense slider ticks aren’t used at all, let alone for countermelodies. An interesting idea for catch notes is as a more readable replacement for accelerating sliders, or to telegraph confusing slider movement.
Anyways, the main question here is how could low intensity sections be mapped, while still keeping a strong connection to the music and not killing flow. While I am partial to catch notes because of my rhythm gaming background and belief that they are more intuitive, the initially proposed weak hit objects or even something else could fill the role, like bomb notes that must be avoided.

**Reply by LocalGuard039** [reply]
Those two are not. 

Just to be clear, touch notes actually have strict judgement from what I see, touch to collect gave player ability to hit early but itself and its hitsound are timed strictly. 

And you can see almost all(or just all?) touch note exist because player have the ability to hit multiple targets at one time. A touch note without proper guide would only cause unwanted hitresult, unwanted off beat sound or stun player's movement in that area and so on, especially in this single thread game if not the sliderbody then you likely would confuse, not to say if this very single note at open area I as well hit it, if it exist at crowded area it breaks alternative again as well, better hit it. The guidance problem is too critical that touch outside of slider would make people confuse when to hit and ruin the gameplay, though.

And like what if a touch note exist between two strong objects? The actual gameplay would be a sliderhead without sliderbody but with tick and another object hidden under this unseen sliderbody, you even have no idea when to catch this tick. Long or short. You can see touch note is slider tick or fast reverse, with same strict judgement, same touch to collect function with some guidance that it is easy to collect in most situation and always hit sync to the beat.

**Comment by cl8n** [comment]
while I think this would *work* in that it fixes the now-missing choice of leniency in mapping and isn't impossible to communicate well to the player... I don't think it can replicate the magic of stable's sliders

imo the whole psychology thing of how sliders "feel" has a lot more going on than just the timing leniency, that's just one of the big factors. there's also the aim leniency in the follow circle as you complete it, and that it corrects your timing passively with the ticks and ends both having visual and audio cues despite no timed inputs. the complete mechanics of sliders and their typical usage in lower density patterns all combine to create that "certain"/"safe" association, it's not something that can be created well by just one component. I guess I'd like to be proven wrong, though

respect the effort (and the essays...) but in the end this doesn't feel better to me than just sliders being the "weak objects" and circles staying the same. which is a shame because that option being rejected in lazer (or doomed to Classic mod) must have helped prompt your idea in the first place. I wish we could just go back on this one, slider leniency will always stick with me as one of the most genius accidents in game design, and its removal is one of my strongest disagreements about this whole project
